router/qrouter/proxy_routing.go

```diff
@@ -1009,10 +1009,13 @@ func (qr *ProxyQrouter) routeWithRules(ctx context.Context, stmt lyx.Node, sph s
 		 */
 		has_inf_schema := false
 		only_catalog := true
+		any_catalog := false
 		has_other_schema := false
 		for rqfn := range meta.rels {
 			/* schema can be omittted here, check for pg_ prefix. */
-			if len(rqfn.RelationName) < 3 || rqfn.RelationName[0:3] != "pg_" {
+			if len(rqfn.RelationName) >= 3 && rqfn.RelationName[0:3] != "pg_" {
```

Comment:
```suggestion
			if len(rqfn.RelationName) >= 3 && rqfn.RelationName[0:3] == "pg_" {
```

Reply:
only_catalog && any_catalog leads to confusion